color_picker/src/main/ets/components/MainPage/colorPicker.ets

```diff
@@ -0,0 +1,216 @@
+            this.SliderL = 50
+            this.s = 100
+            this.l = 50
+            var a = this.HSLToRGB(this.h, this.SliderS, this.SliderL)
+            var b = this.rgbToHex(a[0], a[1], a[2])
+            this.col = b
+            this.ColorFromGradient = b
+            this.ColorFromS = b
+          })
+        Row()
+          .height(40)
+        Flex({ justifyContent: FlexAlign.Start }) {
+          Text("Saturation :").fontWeight(FontWeight.Bolder).fontSize(15)
+        }
+
+        Row({ space: 5 }) {
+          Slider({
+            value: this.SliderS,
+            min: 0,
+            max: 100,
+            style: SliderStyle.OutSet
+          })
+            .blockColor(Color.Blue)
+            .trackColor(Color.Gray)
+            .selectedColor(Color.Blue)
+            .showSteps(true)
+            .showTips(true)
+            .trackThickness(5)
+            .onChange((value: number, mode: SliderChangeMode) => {
+              this.SliderS = value
+              var a = this.HSLToRGB(this.h, this.SliderS, this.SliderL)
+              var b = this.rgbToHex(a[0], a[1], a[2])
+              this.ColorFromS = b
+              this.col = b
+            })
+        }
+        .width('100%')
+
+        Row()
+          .width(340)
+          .height('5%')
+          .borderRadius(20)
+          .linearGradient({
+            angle: 90,
+            colors: [[0x808080, 0], [this.ColorFromGradient, 1]]
+          })
+        Row()
+          .height(40)
+        Flex({ justifyContent: FlexAlign.Start }) {
+          Text("Lightness :").fontWeight(FontWeight.Bolder).fontSize(15)
+        }
+
+        Row({ space: 5 }) {
+          Slider({
+            value: this.SliderL,
+            min: 0,
+            max: 100,
+            style: SliderStyle.OutSet
+          })
+            .blockColor(Color.Blue)
+            .trackColor(Color.Gray)
+            .selectedColor(Color.Blue)
+            .showSteps(true)
+            .showTips(true)
+            .trackThickness(5)
+            .onChange((value: number, mode: SliderChangeMode) => {
+              this.SliderL = value
+              var a = this.HSLToRGB(this.h, this.SliderS, this.SliderL)
+              var b = this.rgbToHex(a[0], a[1], a[2])
+              this.col = b
+            })
+        }
+        .width('100%')
+
+        Row()
+          .width(340)
+          .height('5%')
+          .borderRadius(20)
+          .linearGradient({
+            angle: 90,
+            colors: [[Color.Black, 0], [this.ColorFromS, 1 / 2], [Color.White, 1]]
+          })
+        Row()
+          .height(30)
+        Row({ space: 15 }) {
+          Text("Hex Color Code:").fontWeight(FontWeight.Bolder).fontSize(20)
+          Text(this.col).fontWeight(FontWeight.Bolder).fontColor(this.col).fontSize(25)
+        }
+      }
+      .padding({ right: '2%', left: '2%', top: '1%' })
+    }
+    .opacity(this.OpacityOne)
+    .height('100%')
+  }
+}
+
+namespace colorPicker {
```

Comment:
Why is this empty class required ?

Reply:
Actually, I created a model for ColorPicker Screen by this namespace export which I implemented in ColorPickerUI.ets, and since there is no variable required to be changed by user, so I passed no variable here.